api/lib/domain/events/handle-complementary-certifications-scoring.js

```diff
@@ -30,60 +33,153 @@ async function handleComplementaryCertificationsScoring({
+      }
+      return await _saveCurrentLevel(
+        complementaryCertificationCourseResultRepository,
+        complementaryCertificationScoring,
+      );
+    } else {
+      return await _saveCurrentLevel(
+        complementaryCertificationCourseResultRepository,
+        complementaryCertificationScoring,
+      );
     }
+  }
 
+  async function _saveLowerLevel({ complementaryCertificationScoring, badgeNextLowerLevel }) {
     await complementaryCertificationCourseResultRepository.save(
       ComplementaryCertificationCourseResult.from({
-        ...complementaryCertificationScoringWithComplementaryReferential,
+        ...complementaryCertificationScoring,
+        complementaryCertificationBadgeId: badgeNextLowerLevel.id,
         source: ComplementaryCertificationCourseResult.sources.PIX,
-        acquired: complementaryCertificationScoringWithComplementaryReferential.isAcquired(),
+        acquired: true,
       }),
     );
   }
 }
 
-function _buildComplementaryCertificationScoringWithReferential(
+async function _saveCurrentLevel(complementaryCertificationCourseResultRepository, complementaryCertificationScoring) {
+  await complementaryCertificationCourseResultRepository.save(
+    ComplementaryCertificationCourseResult.from({
+      ...complementaryCertificationScoring,
+      source: ComplementaryCertificationCourseResult.sources.PIX,
+      acquired: complementaryCertificationScoring.isAcquired(),
+    }),
+  );
+}
+
+function _hasAcquiredLowerLevel({
+  pixScore,
+  reproducibilityRate,
+  minimumEarnedPixForCurrentLevel,
+  minimumEarnedPixForLowerLevel,
+  complementaryCertificationScoring,
+  minimumReproducibilityRateLowerLevel,
+}) {
+  if (pixScore < minimumEarnedPixForLowerLevel) {
+    return false;
+  }
+  if (reproducibilityRate < minimumReproducibilityRateLowerLevel) {
+    return false;
+  }
+  return (
+    (_isBelowMinimumCurrentLevelScore({ pixScore, minimumEarnedPixForCurrentLevel }) &&
+      _isAboveMinimumLowerLevelScore({ pixScore, minimumEarnedPixForLowerLevel })) ||
+    (_isBelowMinimumReproducibilityRate(complementaryCertificationScoring) &&
+      _isAboveMinimumLowerReproducibilityRate(complementaryCertificationScoring, minimumReproducibilityRateLowerLevel))
+  );
+}
+
+function _isBelowMinimumReproducibilityRate(complementaryCertificationScoring) {
+  return (
+    complementaryCertificationScoring.reproducibilityRate.value <
+    complementaryCertificationScoring.minimumReproducibilityRate
+  );
+}
+
+function _isAboveMinimumLowerReproducibilityRate(
+  complementaryCertificationScoring,
+  minimumReproducibilityRateLowerLevel,
+) {
+  return complementaryCertificationScoring.reproducibilityRate.value >= minimumReproducibilityRateLowerLevel;
+}
+
+function _isBelowMinimumCurrentLevelScore({ pixScore, minimumEarnedPixForCurrentLevel }) {
+  return pixScore < minimumEarnedPixForCurrentLevel;
+}
+
+function _isAboveMinimumLowerLevelScore({ pixScore, minimumEarnedPixForLowerLevel }) {
```

Comment:
À la lecture, l'intention ne m'a pas parue évidente : 

```suggestion
function _isAboveMinimumLowerLevelRequiredPixScore({ pixScore, minimumEarnedPixForLowerLevel }) {

```